### ISSUE TITLE
Nuclear repulsion energy is tough to find!

### ISSUE THREAD

**Issue by Cryoris** [Qiskit/qiskit-nature#679]
### What is the expected enhancement?

It's pretty difficult to find the nuclear repulsion energy (and probably some other interesting properties) if one doesn't already know where to look:
```python
# es_problem is an `ElectronicStructureProblem`
es_problem.grouped_property_transformed.get_property("ElectronicEnergy").nuclear_repulsion_energy
```

It would be great it the tutorials would mention where they are, e.g. [the ground state solver tutorial](https://github.com/Qiskit/qiskit-nature/blob/main/docs/tutorials/03_ground_state_solvers.ipynb).

(Originated from [this Slack convo](https://qiskit.slack.com/archives/CB6C24TPB/p1653900859430849))

**Comment by mrossinek** [comment]
You are right! The [properties tutorial](https://qiskit.org/documentation/nature/tutorials/08_property_framework.html) should cover this but it does not specifically talk about the nuclear repulsion energy. In fact, the string-representation of the `ElectronicEnergy` class should be improved to also include this attribute. This requires implementing the `ElectronicEnergy.__str__` method.

I do not think that this information should go into the GS tutorial since what you are suggesting is specifically designed for extracting the information **before** running a solver.
Instead, a logical link should be made between the [electronic structure tutorial](https://qiskit.org/documentation/nature/tutorials/01_electronic_structure.html) and the aforementioned properties tutorial.


To summarize, fixing this issue involves the following tasks:
- [ ] implement `ElectronicEnergy.__str__` which should:
    - [ ] call `super().__str__()`
    - [ ] append information on `self.nuclear_repulsion_energy` and `self.reference_energy`
    - [ ] append information on `self.orbital_energies`
    - [ ] append information on `self.kinetic` and `self.overlap`
- [ ] add a section to the Electronic Structure tutorial highlighting how to extract properties from the `ElectronicStructureProblem`:
    - [ ] showcase `print(es_problem)`
    - [ ] i.e. `es_problem.get_property("some property name")`
    - [ ] link to the properties tutorial for more information

**Comment by woodsp-ibm** [comment]
I think too that we should make it clear that what is computed is the Electronic Energy only. Some users expect it to be the total energy and don't get the value they expect and their dissociation profile is not as they expect. I often have to point this fact out then explain to add in the nuclear repulsion energy and where to get it comes next. Of course when transforms are in place things can be more complicated when someone tries to do things themselves through the stack. But the above its the basics which many people seem to want to do by themselves ie get the operator from the stack and then run VQE themselves and its the result from there that they then go onto use.

**Comment by amyels** [comment]
Will look into resolving this issue!

**Comment by mrossinek** [comment]
> Will look into resolving this issue!

Awesome! :+1:

**Comment by mrossinek** [comment]
A short update on this: the `ElectronicStructureProblem` and `ElectronicEnergy` are undergoing a major refactoring, of which #796 does a lot of the intended changes.
This means, that my outline above is no longer up-to-date. I will make sure, that as part of this refactoring, we will significantly improve the visibility of `nuclear_repulsion_energy`, fixing this issue.

But I am unassigning and removing the `good first issue` label from here for the time being,
